drivers/gpu/drm/virtio/virtgpu_vq.c

```diff
@@ -1296,3 +1296,18 @@ void virtio_gpu_cmd_set_scanout_blob(struct virtio_gpu_device *vgdev,
 
 	virtio_gpu_queue_ctrl_buffer(vgdev, vbuf);
 }
+
+void virtio_gpu_cmd_set_modifier(struct virtio_gpu_device *vgdev,
+				     uint32_t scanout_id,
+				     struct drm_framebuffer *fb)
+{
+	struct virtio_gpu_set_modifier *cmd_p;
+	struct virtio_gpu_vbuffer *vbuf;
+
+	cmd_p = virtio_gpu_alloc_cmd(vgdev, &vbuf, sizeof(*cmd_p));
```

Comment:
check for allocation failure?

Reply:
virtio gpu driver code cannot handle the allocation failure, so in the code path of  function virtio_gpu_alloc_cmd, we can see finally this memory is allocated by kmem_cache_zalloc with flag __GFP_NOFAIL.